File: swap.py
"""
ID: yscript1
LANG: PYTHON3
TASK: swap
"""
fin = open ('swap.in', 'r')
fout = open ('swap.out', 'w')
n,k = map(int, fin.readline().strip().split())
a1,a2 = map(int, fin.readline().strip().split())
b1,b2 = map(int, fin.readline().strip().split())
cows = [i for i in range(1,n+1)]
print(cows)
def reverse(cows, start, end):
    t = cows[0:start]
    t.extend(cows[start:end+1][::-1])
    t.extend(cows[(end+1):])
    return t

t = cows[:]
for i in range(k):
    t = reverse(t, a1-1,a2-1)
    t = reverse(t, b1-1,b2-1)
    #print(i, t)
print(t)
fout.write('\n')
fout.close()

File: swap.c
/*
ID: yscript1
LANG: C
TASK: swap
*/
#include <stdio.h>
#include <stdlib.h>
#define DEBUG
#ifdef DEBUG
#define PRINTD(fmt, args...)    fprintf(stderr, fmt, ## args)
#else
#define PRINTD(fmt, args...)    /* Don't do anything in release builds */
#endif

void reverse(int *arr, int start, int end) {
    int t;
    while(start < end) {
        t = arr[start];
        arr[start] = arr[end];
        arr[end] = t;
        start ++;
        end --;
    }
}

void dump_cow(int *cows, int n) {
    for(int i=0; i<n; i++) {
        PRINTD("%d ", cows[i]);
    }
    PRINTD("\n");
}

void main() {
    FILE *fin  = fopen ("swap.in", "r");
    FILE *fout = fopen ("swap.out", "w");
    int cows[101];
    int n, k, a1, a2, b1, b2;
    
    fscanf(fin,"%d %d",&n, &k);
    fscanf(fin,"%d %d",&a1, &a2);
    fscanf(fin,"%d %d",&b1, &b2);

    for(int i=0; i<n; i++)
        cows[i] = i + 1;

    for(int i=0; i<k; i++) {
        reverse(cows, a1-1, a2-1);
        /* dump_cow(cows, n); */
        reverse(cows, b1-1, b2-1);
        /* dump_cow(cows, n); */
    }
    
    dump_cow(cows,n);
    
    fclose(fin);
    fclose(fout);
    exit(0);
}